### DISCUSSION TITLE
Swipe up/down and long press on each button

### DISCUSSION THREAD

**Discussion by nigmastar** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#34] [Q&A] [answered]
Hello, currently I know these are not available but I wonder if the following will ever be possible on the TX Ultimate:

1) swipe up/down on each button/gang
2) long press on each button/gang

Having either of the 2 above would be amazing for implementing dimmers on each gang.

Thank you

**Comment by barth11** [comment, marked as answer]
1. Not possible because: https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/discussions/61#discussioncomment-11291182
2. Should be possible. Extending the long press specification to specific touch stripes. (https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/blob/main/components/tx_ultimate_touch/tx_ultimate_touch.cpp Line75) It's basically L1,L2,L3 but only holding longer.

**Reply by nigmastar** [reply]
thanks!
